src/utils.c

```diff
@@ -148,37 +149,37 @@ Datum
 time_to_internal(PG_FUNCTION_ARGS)
 {
 	Datum		val;
-	Oid			type;
+	Oid			type_oid;
 	int64		res;
 
 	if (PG_ARGISNULL(0))
 		PG_RETURN_NULL();
 
 	val = PG_GETARG_DATUM(0);
-	type = PG_GETARG_OID(1);
+	type_oid = PG_GETARG_OID(1);
 
-	res = time_value_to_internal(val, type);
+	res = time_value_to_internal(val, type_oid, false);
 	PG_RETURN_INT64(res);
 }
 
 
 /*	*/
 int64
-time_value_to_internal(Datum time_val, Oid type)
+time_value_to_internal(Datum time_val, Oid type_oid, bool failure_ok)
 {
-	if (type == INT8OID)
+	if (type_oid == INT8OID)
```

Comment:
While you are messing with this function, can you please turn it into a switch statement? The current code is terrible and inefficient :).

Reply:
Fixed